### PR TITLE
CON-4807: remote categories can be mapped by multiple local categories

### PR DESCRIPTION
**Pull request by keulinho**

### CODE REVIEW

File: Bootstrap/Update.php [hunk 1]
```diff
@@ -427,4 +428,42 @@ private function addConnectItemsIndex()
...
+                  `remote_category_id` int(11) NOT NULL,
+                  `local_category_id` int(11) unsigned NOT NULL,
+                  PRIMARY KEY (`remote_category_id`, `local_category_id`),
+                  CONSTRAINT s_plugin_connect_remote_categories_fk_remote_category_id FOREIGN KEY (remote_category_id) REFERENCES s_plugin_connect_categories (id) ON DELETE CASCADE,
+                  CONSTRAINT s_plugin_connect_remote_categories_fk_local_category_id FOREIGN KEY (local_category_id) REFERENCES s_categories (id) ON DELETE CASCADE
+                  ) ENGINE = InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;'
+                );
+                $result = $this->db->query('SELECT pcc.id, pcc.local_category_id
+                                FROM s_plugin_connect_categories pcc
+                                WHERE pcc.local_category_id is not null');
```
Comment (sbofirov) [inline Bootstrap/Update.php]: please use capital letters for MySQL keywords reserved words. In this case ```IS NOT NULL```

File: Bootstrap/Update.php [hunk 2]
```diff
@@ -427,4 +428,42 @@ private function addConnectItemsIndex()
...
+
+                while ($row=$result->fetch()) {
```
Comment (sbofirov) [inline Bootstrap/Update.php]: Missing one space between ```=```
Maybe you need to run ```bash .githooks/install-hooks.sh``` locally. After that before every commit php cs fixer will scan your code and notify you have some places where code doesn't have correct code style.

File: Models/Connect/RemoteCategory.php [hunk 3]
```diff
@@ -97,34 +97,36 @@ public function setLabel($label)
...
-        $this->localCategoryId = $id;
+        return $this->localCategories->toArray();
     }
 
     /**
-     * @return \Shopware\Models\Category\Category
+     * @param \Shopware\Models\Category\Category $localCategory
      */
-    public function getLocalCategory()
+    public function addLocalCategory($localCategory)
```
Comment (sbofirov) [inline Models/Connect/RemoteCategory.php]: Please use typehint. It must be ```public function addLocalCategory(Category $localCategory)``` where Category is Shopware\Models\Category\Category.

File: Models/Connect/RemoteCategory.php [hunk 4]
```diff
@@ -97,34 +97,36 @@ public function setLabel($label)
...
+        if (!$this->localCategories->contains($localCategory)) {
+            $this->localCategories->add($localCategory);
+        }
     }
 
     /**
      * @param \Shopware\Models\Category\Category $localCategory
      */
-    public function setLocalCategory($localCategory)
+    public function removeLocalCategory($localCategory)
```
Comment (sbofirov) [inline Models/Connect/RemoteCategory.php]: same here

File: Bootstrap/Update.php (hunk 2 above)
Comment (keulinho) [reply Bootstrap/Update.php]: the cs fixer ran but i wil fix this

**Pull request merged**